allennlp/sanity_checks/task_checklists/task_suite.py

```diff
@@ -0,0 +1,372 @@
+import sys
+import logging
+from typing import Type, Optional, Dict, Any, Callable, List, Iterable, Union, TextIO
+from checklist.test_suite import TestSuite
+from checklist.editor import Editor
+from checklist.test_types import MFT, INV, DIR
+from checklist.perturb import Perturb
+from allennlp.common.registrable import Registrable
+from allennlp.common.file_utils import cached_path
+from allennlp.predictors.predictor import Predictor
+from allennlp.sanity_checks.task_checklists import utils
+
+logger = logging.getLogger(__name__)
+
+
+class TaskSuite(Registrable):
+    """
+    Base class for various task test suites.
+
+    This is a wrapper class around the CheckList toolkit introduced
+    in the paper
+    [Beyond Accuracy: Behavioral Testing of NLP models with CheckList (Ribeiro et al)]
+    (https://api.semanticscholar.org/CorpusID:218551201).
+
+    Task suites are intended to be used as a form of behavioral testing
+    for NLP models to check for robustness across several general linguistic
+    capabilities; eg. Vocabulary, SRL, Negation, etc.
+
+    An example of the entire checklist process can be found at:
+    https://github.com/marcotcr/checklist/blob/master/notebooks/tutorials/
```

Comment:
Annoyingly, this doesn't seem to be recognized as a link unless we explicitly make it one.

```suggestion
    [https://github.com/marcotcr/checklist/blob/master/notebooks/tutorials/]
    (https://github.com/marcotcr/checklist/blob/master/notebooks/tutorials/).
```